File: code/ex1063.java
class Solution {
    public int validSubarrays(int[] nums) {
        int n = nums.length, res = 0;
        Stack<int[]> stack = new Stack<>();    // {val, idx}
        
        for (int i = 0; i < n; i++) {
            while (!stack.isEmpty() && stack.peek()[0] > nums[i])
                res += i - stack.pop()[1];
            
            stack.push(new int[]{nums[i], i});
        }
        
        while (!stack.isEmpty())
            res += n - stack.pop()[1];

        return res;
    }
}

File: code/ex1063.cpp
class Solution {
public:
    int validSubarrays(vector<int>& nums) {
        int n = nums.size(), cnt = 0;
        
        stack<pair<int, int>> stk;    // Monotonously increasing {val, idx}
        for (int i = 0; i < n; i++) {
            while (!stk.empty() && nums[i] < stk.top().first) {
                cnt += i - stk.top().second;
                stk.pop();
            }
            stk.push({nums[i], i});
        }
        
        // Tackle remaining pairs in the stack
        while (!stk.empty()) {
            cnt += n - stk.top().second;
            stk.pop();
        }
        
        return cnt;
    }
};